gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/ZookeeperRemoteAliasService.java

```diff
@@ -127,9 +156,18 @@ private static void ensureEntry(final String path, final RemoteConfigurationRegi
                     // content may not be trustworthy.
                     if (remoteClient.isAuthenticationConfigured()) {
                         LOG.correctingSuspectWritableRemoteConfigurationEntry(path);
-
                         // Replace the existing ACL with one that permits only authenticated users
-                        remoteClient.setACL(path, Collections.singletonList(AUTHENTICATED_USERS_ALL));
+                        if (remoteClient.authenticationType()
```

Comment:
`.authenticationType()` may be NULL (see `LocalFileSystemRemoteConfigurationRegistryClientService` implementation below) -> `remoteClient.authenticationType().equalsIgnoreCase("Kerberos")` may throw a NPE.

Reply:
` LocalFileSystemRemoteConfigurationRegistryClientService` is used only for testing, that is why it returns null, but it is better to defend agains NPEs, will fix it.